include/dca/parallel/stdthread/thread_pool/thread_pool.hpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2012 Jakob Progsch, Václav Zeman
+// Copyright (C) 2018 ETH Zurich
+// Copyright (C) 2018 UT-Battelle, LLC
+// All rights reserved.
+//
+// See LICENSE for terms of usage.
+// See CITATION.md for citation guidelines, if DCA++ is used for scientific publications.
+//
+// Author: Giovanni Balduzzi (gbalduzz@itp.phys.ethz.ch)
+//
+// This file implements a thread pool based on the file
+// https://github.com/progschj/ThreadPool/blob/master/ThreadPool.h.
+// See COPYING for the license of the original work.
+
+#ifndef DCA_PARALLEL_STDTHREAD_THREAD_POOL_THREAD_POOL_HPP
+#define DCA_PARALLEL_STDTHREAD_THREAD_POOL_THREAD_POOL_HPP
+
+#include <condition_variable>
+#include <iostream>
+#include <functional>
+#include <future>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <stdexcept>
+#include <thread>
+#include <vector>
+
+namespace dca {
+namespace parallel {
+
+class ThreadPool {
+public:
+  // Creates a pool with n_threads.
```

Comment:
It might be a trivial function to document, yet I don't think a description hurts. At the very least it helps visually separating the methods. Anyway I it in b05b9438e6b40e160423ef891f85d20c7bc30a1c